### ISSUE TITLE
No way to specify multiple escape characters

### ISSUE THREAD

**Issue by haltman-at** [ljharb/shell-quote#8]
For some uses, it might be nice to be able to specify multiple characters all of which can act as an escape.  Right now you can only have one escape character, as best I can tell; if you pass a multi-character string as the escape character, the result is that you won't be able to escape at all.

Thanks!

**Comment by ljharb** [comment]
What form would you expect?

I can imagine a regular expression or a predicate function, but those could have unpredictable results - a simpler option is to accept an array of multiple strings, but that would break anyone who'd been passing in an array of a single character. However, I'd hope nobody is doing this.

**Comment by haltman-at** [comment]
Accepting multiple characters as a single string is also a possibility -- a character is an escape if it appears in the string.

**Comment by ljharb** [comment]
true. altho it's probably more likely that someone is using a multi-char escape sequence than that they're passing an array containing one item

**Comment by ljharb** [comment]
Can you elaborate on the use cases?

**Comment by haltman-at** [comment]
The intended use case was to mimic a Windows shell, where both `^` and backtick are escape characters.

**Comment by ljharb** [comment]
Gotcha - this seems related to #10, #21, #25 - and since this package is for POSIX shells, using it for (non-WSL) windows is likely a nonstarter. Thanks for following up!
